libs/platform/src/lib/components/form/platform-file-uploader/platform-file-uploader.component.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { FormsModule } from '@angular/forms';
+import { By } from '@angular/platform-browser';
+import { FdpFormGroupModule } from '../form-group/fdp-form.module';
+
+import { PlatformFileUploaderComponent } from './platform-file-uploader.component';
+import { PlatformFileUploaderModule } from './platform-file-uploader.module';
+
+@Component({
+    selector: 'fdp-mulit-input-test',
+    template: `
+        <fdp-form-group #fg>
+            <fdp-form-field
+                #ffl1
+                label="Default Multi Input Field"
+                id="input-simple"
+                name="reactiveFormInput"
+                zone="zLeft"
+                rank="1"
+                placeholder="Select the file"
+            >
+                <fdp-file-uploader
+                    placeholder="Select the file"
+                    buttonLabel="Browse"
+                    buttonAriaLabel="browse file"
+                    accept=".png,.jpg"
+                    [(ngModel)]="files"
+                    [disabled]="true"
+                    [multiple]="true"
+                    (selectionChange)="handleFileSelection($event)"
+                ></fdp-file-uploader>
+            </fdp-form-field>
+        </fdp-form-group>
+    `
+})
+class PlatformFileUploaderTest {
+    @ViewChild(PlatformFileUploaderComponent)
+    platformfileUploader: PlatformFileUploaderComponent;
+}
+
+interface MockFile extends File {
+    size: number;
+}
+
+describe('PlatformFileUploaderComponent', () => {
+    let component: PlatformFileUploaderTest;
+    let fixture: ComponentFixture<PlatformFileUploaderTest>;
+    let fileinput: PlatformFileUploaderComponent;
+
+    beforeEach(async () => {
+        await TestBed.configureTestingModule({
+            declarations: [PlatformFileUploaderComponent, PlatformFileUploaderTest],
+            imports: [CommonModule, FormsModule, FdpFormGroupModule, PlatformFileUploaderModule]
+        }).compileComponents();
+    });
+
+    beforeEach(() => {
+        fixture = TestBed.createComponent(PlatformFileUploaderTest);
+        component = fixture.componentInstance;
+        fixture.detectChanges();
+        fileinput = component.platformfileUploader;
+    });
+    async function wait(componentFixture: ComponentFixture<any>): Promise<void> {
+        componentFixture.detectChanges();
+        await componentFixture.whenStable();
+    }
+
+    it('should create', () => {
+        expect(component).toBeTruthy();
+    });
+
+    it('should setDisabledState', async () => {
+        await wait(fixture);
+        fileinput.disabled = true;
+        fixture.detectChanges();
+        const multiInputData = fixture.debugElement.query(By.css('fd-file-uploader'));
+        const disabledValue = multiInputData.nativeElement.getAttribute('ng-reflect-disabled');
+        expect(disabledValue).toBe('true');
+    });
+
+    it('should select file upload', async () => {
+        await wait(fixture);
```

Comment:
can we have few more tests that test the working of file uploader with platform forms?